### PR TITLE
Enhance Documentation

### PR DESCRIPTION
**Pull request by tasansal**
* Refactored some abstractions (removed some)
* Added much nicer docs
* Added most classes and schemas to docs

Rendered docs with updates here: https://segy--28.org.readthedocs.build/en/28/

### CODE REVIEW

File: docs/data_models/trace.md [hunk 1]
```diff
@@ -33,9 +32,7 @@
 
 ## Trace Header
 
-```{eval-rst}
-.. autopydantic_model:: TraceHeaderDescriptor
-```
+Structured bla bla
```
Comment (ta-hill) [inline docs/data_models/trace.md:35]: Did you want to leave this line in here?

File: docs/settings.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+It is composed of various sub-settings isolated by SEG-Y components and various topics.
+
+- **BINARY**: The [SegyBinaryHeaderSettings] is used for binary header configuration
+  while reading a SEG-Y file.
+- **ENDIAN**: This setting determines the byte order that is being used in the SEG-Y file.
+  The possible options are `"big"` or `"little"` based on [Endianness]. If left as None,
+  the system defaults to Big Endian (`"big"`).
+- **REVISION**: This setting is used to specify the SEG-Y revision number. If left as
+  None, the system will automatically use the revision mentioned in the SEG-Y file.
+- **USE_PANDAS**: This setting is a boolean that decides whether to use pandas for
```
Comment (ta-hill) [inline docs/settings.md:33]: Maybe we emphasize that this only applies to the Trace Headers and not the Trace Data. Also we don't at the moment support turning this setting off.

File: docs/data_models/trace.md (hunk 1 above)
Comment (tasansal) [reply docs/data_models/trace.md:35]: haha no, lemme fix it. thx.

**Review by ta-hill — COMMENTED** [review]
I added a couple of comments for some context in some places, everything else looks good to me

File: docs/settings.md (hunk 2 above)
Comment (tasansal) [reply docs/settings.md:33]: good point i'll fix

**Review by ta-hill — APPROVED** [review]

**Pull request merged**